sources/whippet/src/broker.rs

```diff
@@ -0,0 +1,138 @@
+//! D-Bus broker management and lifecycle control.
+//!
+//! This module provides the [`BrokerManager`] which handles spawning and communicating
+//! with the dbus-broker process, including policy transfer and systemd integration.
+
+use crate::child::BrokerChild;
+use crate::dbus_policy::DbusPolicy;
+use crate::error::{self, Result};
+use crate::socket;
+use snafu::ResultExt;
+use std::os::unix::io::BorrowedFd;
+use std::os::unix::net::UnixStream;
+use zbus::conn::Builder;
+use zbus::proxy;
+use zbus::zvariant::{Fd, ObjectPath, Value};
+
+/// A proxy client used to interact with the spanwed dbus broker once it establishes a connection
+/// back to whippet
+#[proxy(
+    interface = "org.bus1.DBus.Broker",
+    default_path = "/org/bus1/DBus/Broker"
+)]
+trait DBusBroker {
+    /// Implements the AddListener call to transfer the systemd socket and dbus policy to the
+    /// broker
+    fn add_listener(&self, path: ObjectPath<'_>, fd: Fd<'_>, policy: Value<'_>)
+        -> zbus::Result<()>;
+}
+
+/// Manages the lifecycle of the spawned broker
+pub(crate) struct BrokerManager {
+    child: BrokerChild,
+    controller: UnixStream,
+    policy: DbusPolicy,
+}
+
+impl BrokerManager {
+    pub(crate) async fn build_with(broker_path: &str, policy: DbusPolicy) -> Result<Self> {
+        debug!("Creating BrokerManager with broker path: {}", broker_path);
+
+        debug!("Creating controller socket pair");
+        let (controller, broker_end) = socket::controller_pair()?;
+
+        debug!("Creating journal socket");
+        let journal_socket = socket::journal_socket()?;
+
+        // Transfer ownwership of the sockets to the child process, so that they are cleaned up
+        // automatically once the child goes out of scope (no extra drop calls required)
+        let child = BrokerChild::new(broker_path, broker_end, journal_socket)?;
+
+        Ok(Self {
+            child,
+            controller,
+            policy,
+        })
+    }
+
+    /// Spawn the broker and transfer the systemd socket and policy
+    pub(crate) async fn run(&mut self) -> Result<()> {
+        // Start the broker in the background but don't block on it being ready, as the AddListener
+        // call will block until the broker is ready to receive calls in the controller socket
+        self.child.run()?;
+        debug!("Adding listener with parsed policy");
+        self.add_listener().await?;
+        info!("Policy applied successfully");
+        // Notify the broker is ready once the policy and the sockets were passed to the broker.
+        // This deviates slightly from the dbus-launcher as it notifies systemd right after the
+        // configuration is loaded
+        self.notify_systemd_ready()?;
+        info!("Notified systemd that service is ready");
+
+        // Wait forenver until signaled to stop or the dbus-broker is killed
```

Comment:
nit: or just omit this comment too
```suggestion
        // Wait forever until signaled to stop or the dbus-broker is killed
```